### DISCUSSION TITLE
Whether can not use allotment even in next.js NOSSR component?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [johnwalley/allotment#657] [Q&A] [answered]
Hi, could this library used in Next.js nossr component? 

I check the description of [allotment](https://www.npmjs.com/package/allotment) 

> Next.js
If you get an error when importing allotment in a Next.js project consider [not including the module server-side](https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr). Allotment currently only works in a browser. It might be possible to produce sensible results server-side in the future so create an issue requesting this if interested.

base on it I think nossr page can use it, so in my next.js with electron template application I created a page like this, I use dynamic import to avoid SSR:

```
import { Allotment } from "allotment";
import "allotment/dist/style.css";

import dynamic from 'next/dynamic'

const Layout = dynamic(() => import('../components/Layout'), {
    ssr: false,
})


const IndexPage = () => {

  return (
    <Layout>
      start 
      <Allotment>
        <Allotment.Pane minSize={200}>
          <div style={{ width: 200, height: 200 }}>div1</div>
        </Allotment.Pane>
        <Allotment.Pane snap  minSize={200}>
          <div style={{ width: 200, height: 200 }}>div2</div>
        </Allotment.Pane>
      </Allotment>
      end
    </Layout>
  )
}

export default IndexPage

```

but it doesn't seem to be working:

![image](https://github.com/johnwalley/allotment/assets/12727582/3e723084-a18c-4464-bc7f-32fa603a43cd)

So in there I want to ensure whether disable SSR in next.js also cannot use allotment.

**Comment by johnwalley** [comment, marked as answer]
I wonder if it's because the `Allotment` component has no intrinsic height so it requires its parent to have height. For example, if you were to wrap the `Allotment` component in a `div` with a non-zero height does it help?
